hack/generator/pkg/codegen/code_generator.go

```diff
@@ -126,6 +126,11 @@ func createAllPipelineStages(idFactory astmodel.IdentifierFactory, configuration
 
 		stripUnreferencedTypeDefinitions(),
 
+		// TODO: We need better testing for the below
+		removeEmbeddedResources().UsedFor(ArmTarget), // TODO: For now only used for ARM,
+
+		removeEmptyObjectTypes(),
```

Comment:
Empty objects in the JSON schema currently get turned into `map[string]v1.JSON`.

Since it's not actually possible to have an empty object type from the JSON schema, the only way that this can happen is as part of `removeEmbeddedResources()`. Wondering if it makes sense to just push this into that stage?

Reply:
This means I can't refactor it into a separate PR though. I'll go ahead and combine them though since there seems to be agreement that would be better.